### PR TITLE
Consistent interface and various improvements for pycolmap/estimators

### PR DESCRIPTION
**Pull request by ahojnnes**
With a best effort of raising deprecation warnings for  renamed function names. For example:
```
>>> pycolmap.essential_matrix_estimation()
<stdin>:1: DeprecationWarning: essential_matrix_estimation() is deprecated, use estimate_essential_matrix() instead.
...
```
Other modules follow in separate PRs.

### CODE REVIEW

**Review by B1ueber2y — APPROVED** [review]
Looks great! Thanks for the DefDeprecation feature. Since this affects almost all the HLoc interfaces, I suggest to wait for Paul to give a feedback.

File: src/pycolmap/helpers.h [hunk 1]
```diff
@@ -406,3 +407,28 @@ inline bool IsPyceresAvailable() {
...
+    std::optional<std::string> custom_warning = std::nullopt) {
+  parent.def(
+      old_name.c_str(),
+      [parent,
+       old_name = std::move(old_name),
+       new_name = std::move(new_name),
+       custom_warning = std::move(custom_warning)](const py::args& args,
+                                                   const py::kwargs& kwargs) {
+        if (custom_warning) {
+          PyErr_WarnEx(PyExc_DeprecationWarning, custom_warning->c_str(), 1);
```
Comment (sarlinpe) [inline src/pycolmap/helpers.h:425]: Does this print every single time it is called? if so, should we use `LOG_FIRST_N(WARNING, N)` instead?

File: src/pycolmap/estimators/similarity_transform.cc [hunk 2]
```diff
@@ -54,8 +54,9 @@ void BindSimilarityTransformEstimator(py::module& m) {
                         "num_inliers"_a = report.support.num_inliers,
                         "inliers"_a = ToPythonMask(report.inlier_mask));
```
Comment (sarlinpe) [inline src/pycolmap/estimators/similarity_transform.cc:55]: Should this be `inlier_mask` to be consistent with the other changes?

File: src/pycolmap/helpers.h (hunk 1 above)
Comment (ahojnnes) [reply src/pycolmap/helpers.h:425]: No, Python's deprecation warnings only get issued once upon first usage, so it is equivalent to LOG_FIRST_N() with the added benefit of working nicely together with Python's warning filtering system, etc.

File: src/pycolmap/estimators/similarity_transform.cc (hunk 2 above)
Comment (ahojnnes) [reply src/pycolmap/estimators/similarity_transform.cc]: Good catch. Done.

**Pull request merged**